ouroboros-network/src/Ouroboros/Network/Mux/Control.hs

```diff
@@ -0,0 +1,74 @@
+{-# LANGUAGE BangPatterns   #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE TypeFamilies   #-}
+
+module Ouroboros.Network.Mux.Control where
+
+import qualified Codec.CBOR.Decoding as CBOR
+import qualified Codec.CBOR.Encoding as CBOR
+import qualified Codec.CBOR.Read as CBOR
+import qualified Codec.CBOR.Term as CBOR
+import           Codec.CBOR.Write (toLazyByteString)
+import           Codec.Serialise.Class
+import           Control.Exception
+import           Control.Monad
+
+import qualified Data.Map.Strict as M
+import           Data.Maybe (mapMaybe)
+import           GHC.Stack
+
+import           Ouroboros.Network.Mux.Types
+
+data ControlMsg = MsgInitReq [Version]
+                | MsgInitRsp Version
+                | MsgInitFail String
+                deriving Show
+
+encodeCtrlMsg :: ControlMsg -> CBOR.Encoding
```

Comment:
Do you have a test that validates `decoder . encoder = id`? If not maybe add an issue, so we don't forget about it.

Reply:
I don't but it is needed.

Reply:
Implemented.